conans/client/tools/win.py

```diff
@@ -364,36 +364,10 @@ def vcvars_command(settings, arch=None, compiler_version=None, force=False, vcva
         last_version = latest_vs_version_installed(output=output)
 
         compiler_version = compiler_version or last_version
-    os_setting = settings.get_safe("os")
     if not compiler_version:
         raise ConanException("compiler.version setting required for vcvars not defined")
 
-    # https://msdn.microsoft.com/en-us/library/f2ccy3wt.aspx
```

Comment:
Why this movement of code? Please, revert it, it is hard to see what you changed in the diff and we would like to see only the "effective" code.

Reply:
Maybe you can just introduce an "if" at the beginning and returning.

Reply:
I extracted the code to make it testable for visual studio versions, which are not installed in the CI build infrastructure...
I adapted the unit test and reverted the extracting.